tests/product/test_connectors.py

```diff
@@ -43,30 +42,33 @@ def setup_cluster_assert_connectors(self):
         self._assert_connectors_loaded([['system'], ['tpch']])
 
     @attr('smoketest')
-    def test_basic_connector_add_remove(self):
+    def test_connector_add_remove(self):
         self.setup_cluster_assert_connectors()
-
         self.run_prestoadmin('connector remove tpch')
-        self.run_prestoadmin('server restart')
         self.assert_path_removed(self.cluster.master,
                                  os.path.join(constants.CONNECTORS_DIR,
                                               'tpch.properties'))
-        self._assert_connectors_loaded([['system']])
-        for host in self.cluster.all_hosts():
```

Comment:
I say we keep this for loop, it's not that expensive. I agree that we should remove the `server restart` and the `_assert_connectors_loaded` because those test that Presto picks up connectors. But checking that the file was removed on the file system on all nodes should be part of this test.